neqo-qpack/src/decoder.rs

```diff
@@ -479,15 +478,19 @@ impl QPackDecoder {
         if cap > u64::from(self.max_table_size) {
             return Err(Error::EncoderStreamError);
         }
-        self.table.set_capacity(cap);
+        if self.table.set_capacity(cap).is_err() {
+            return Err(Error::EncoderStreamError);
+        }
         Ok(())
     }
 
     fn header_ack(&mut self, stream_id: u64, required_inserts: u64) {
         let ack_increment_delta = required_inserts - self.table.get_acked_inserts_cnt();
         self.send_buf
             .encode_prefixed_encoded_int(0x80, 1, stream_id);
-        self.table.increment_acked(ack_increment_delta);
+        if self.table.increment_acked(ack_increment_delta).is_err() {
+            panic!("This should never happen");
+        }
```

Comment:
Use `x.unwrap("This should never happen")` instead of `if x.is.err() {panic!()}`